### DISCUSSION TITLE
What should `SELECT auth.uid()` return if it is run through SQL Editor

### DISCUSSION THREAD

**Discussion by fenandosr** [supabase/supabase#6997] [Questions] [answered]
As of today, it is returning NULL for me.

I'm trying directly on the SQL Editor (and on Azure Data Studio, which I have been using with no problem).

`SELECT current_user` gives me "supabase_admin", as expected.

Has the behaviour of `auth.uid()` changed recently? Is it possible that this function before gave me my personal authenticated UUID?

**Comment by GaryAustin1** [comment, marked as answer]
It should have always returned NULL if a call did not come in from the API.  Auth.uid() looks at the jwt information to get the uuid.  I believe you can fake out the jwt information with the sql editor before you run auth.uid(), but would have to look up how to do that.

**Reply by fenandosr** [reply]
My use case is very simple:

I use `auth.uid()` as the default value of the column `created_by`, set by a trigger that runs when a record is inserted. This is working fine when using the API, but sometimes we need to seed the database from SQL files: in this case the trigger is failing because `auth.uid()` is NULL.

We have a solution, though.
Thanks

**Reply by bombillazo** [reply]
What was the solution? It would be nice to share back if you were able to get help and sole the issue.

**Reply by Sicria** [reply]
@fenandosr What was your solution for this?

**Reply by fenandosr** [reply]
@bombillazo @Sicria The following function is created as a trigger before insert:

```sql
--
-- On record creation: who and when
--
CREATE OR REPLACE FUNCTION some_scheme.trg_audit_record_created()
RETURNS TRIGGER
LANGUAGE plpgsql AS
$$
BEGIN
  -- Is authenticated?
  SELECT COALESCE(private_schema.tenant_id(), NEW.tenant_id), COALESCE(user_email(), CURRENT_USER)
    INTO NEW.tenant_id, NEW.created_by;
  RETURN NEW;
END;
$$

--
-- User email
--
CREATE OR REPLACE FUNCTION user_email()
RETURNS VARCHAR
LANGUAGE plpgsql SECURITY DEFINER AS
$$
BEGIN
  RETURN ( SELECT email FROM auth.users WHERE id=auth.uid() );
END;
$$

--
-- Users tenant id
--
CREATE OR REPLACE FUNCTION private_schema.tenant_id()
RETURNS UUID
LANGUAGE plpgsql SECURITY DEFINER AS
$$
BEGIN
  RETURN ( SELECT tenant_id FROM sys_users WHERE _id=auth.uid() AND is_active );
END;
$$
```

Note that using `COALESCE(** whatever with auth.uid() **, CURRENT_USER)` is an arbitrary assumption... but within the scope of the use case it can be handled (e.g. RLS).
